src/parameterizations/lateral/MOM_self_attr_load.F90

```diff
@@ -0,0 +1,278 @@
+  type(unit_scale_type),  intent(in)    :: US   !< A dimensional unit scaling type
+  type(param_file_type),  intent(in)    :: param_file !< A structure to parse for run-time parameters.
+  type(SAL_CS), intent(inout) :: CS   !< Self-attraction and loading control structure
+
+  ! Local variables
+# include "version_variable.h"
+  character(len=40)  :: mdl = "MOM_self_attr_load" ! This module's name.
+  integer :: lmax ! Total modes of the real spherical harmonics [nondim]
+  real :: rhoW    ! The average density of sea water [R ~> kg m-3].
+  real :: rhoE    ! The average density of Earth [R ~> kg m-3].
+
+  logical :: calculate_sal
+  logical :: tides, use_tidal_sal_file
+  real :: tide_sal_scalar_value
+
+  ! Read all relevant parameters and write them to the model log.
+  call log_version(param_file, mdl, version, "")
+
+  call get_param(param_file, '', "TIDES", tides, default=.false., do_not_log=.True.)
+  call get_param(param_file, mdl, "CALCULATE_SAL", calculate_sal, "If true, calculate "//&
+                 " self-attraction and loading.", default=tides, do_not_log=.True.)
+  if (.not. calculate_sal) return
+
+  if (tides) then
+    call get_param(param_file, '', "USE_PREVIOUS_TIDES", CS%use_tidal_sal_prev, &
+                   default=.false., do_not_log=.True.)
+    call get_param(param_file, '', "TIDAL_SAL_FROM_FILE", use_tidal_sal_file, &
+                   default=.false., do_not_log=.True.)
+  endif
+
+  call get_param(param_file, mdl, "SAL_SCALAR_APPROX", CS%use_sal_scalar, &
+                 "If true, use the scalar approximation to calculate self-attraction and"//&
+                 " loading.", default=tides .and. (.not.use_tidal_sal_file))
+  call get_param(param_file, mdl, "SAL_HARMONICS", CS%use_sal_sht, &
+                 "If true, use the online spherical harmonics method to calculate"//&
+                 " self-attraction and loading.", default=.false.)
+
+  if (CS%use_sal_scalar .or. CS%use_tidal_sal_prev) then
+    call get_param(param_file, '', "TIDE_SAL_SCALAR_VALUE", tide_sal_scalar_value, &
+                   units="m m-1", default=0.0, do_not_log=.True.)
+    if (tide_sal_scalar_value/=0.0) &
+      call MOM_error(WARNING, "TIDE_SAL_SCALAR_VALUE is a deprecated parameter. "//&
+                              "Use SAL_SCALAR_VALUE instead." )
+      call get_param(param_file, mdl, "SAL_SCALAR_VALUE", CS%sal_scalar_value, &
+                   "The constant of proportionality between sea surface "//&
+                   "height (really it should be bottom pressure) anomalies "//&
+                   "and bottom geopotential anomalies. This is only used if "//&
+                   "USE_SAL_SCALAR is true or USE_PREVIOUS_TIDES is true.", &
+                   default=tide_sal_scalar_value, units="m m-1", &
+                   do_not_log=(.not. CS%use_sal_scalar) .and. (.not. CS%use_tidal_sal_prev))
+  endif
+
+  if (CS%use_sal_sht) then
+    call get_param(param_file, mdl, "SAL_HARMONICS_DEGREE", CS%sal_sht_Nd, &
```

Comment:
> Because of the do_not_log arguments, these 3 get_param calls can be moved outside of the `if (CS%use_sal_sht) then` test.

If I understand correctly. Without the `if` test, these parameters wouldn't be logged when the switch is off, but also wouldn't be marked as "unused". Is it going to be a problem?

Reply:
Your understanding is correct.  The question comes down to how we want to handle the case where the SAL is used and one of these otherwise valid parameters is in an input file (perhaps because of shared input files across sensitivity experiments).  My preference is to allow this, but others (e.g. Alistair) would disagree from time to time.  Perhaps it all comes down to your most recent cause for an annoying model failure.
It is your call on this one, but we don't need both the do_not_log argument and for the calls to be inside of the if block, because as written they are duplicative, if benign.